bubuku/cli.py

```diff
@@ -92,37 +100,34 @@ def trigger_preferred_replica_election(dry_run: bool, max_json_size: int):
                 _LOG.info("Found incorrect assignment: %s, leader is %d, but should be the first one in %s",
                           key, leader, replica_list)
                 wrong_assignment.append(key)
-        while wrong_assignment:
-            items_to_take = len(wrong_assignment)
-            change_applied = False
-            while not change_applied:
-                json_element = {
-                    "version": 1,
-                    "partitions": [{'topic': v[0], 'partition': int(v[1])} for v in wrong_assignment[:items_to_take]]
-                }
-
-                res = json.dumps(json_element).encode('utf-8')
-                if len(res) > max_json_size:
-                    new_items_to_take = int(items_to_take / 2)
-                    _LOG.info("Not fitting to %d bytes with %d items, will try %d items",
-                              max_json_size, items_to_take, new_items_to_take)
-                    items_to_take = new_items_to_take
-                    if items_to_take <= 0:
-                        _LOG.error("Incorrect configuration - even one key is not fitting to proposed size %d. "
-                                   "Stop playing and do the job!", max_json_size)
-                        exit(1)
-                    continue
-                if dry_run:
-                    print(res.decode('utf-8'))
-                else:
+
+        if dry_run:
+            print(_dump_replica_assignment_as_json(wrong_assignment))
```

Comment:
Wait, looks like here lurks a bug with binary vs. str.

Reply:
```
>>> print(b'x')
b'x'
```

Reply:
maybe it is better to do encoding / decoding when needed ? _dump_replica_assignment_as_json could return string, and where needed the string is encoded / decoded

Reply:
Thought about it.  Let me check if we win anything with that.

Reply:
if you return encoded string then maybe we can change method name to make it clear, because I do not expect I have to decode something when I get the value